shell/server/proxy.js

```diff
@@ -310,14 +321,27 @@ function openSessionInternal(grainId, user, title, apiToken) {
 
   updateLastActive(grainId, userId);
 
-  var isOwner = grainInfo.owner === userId;
+  cachedSalt = cachedSalt || Random.id(22);
+  var sessionId = generateSessionId(grainId, userId, cachedSalt);
+  var session = Sessions.findOne({_id: sessionId});
```

Comment:
D'oh. I was somehow forgetting the fact that the salt is randomly generated by the client.